services/collectd/service.go

```diff
@@ -1,11 +1,241 @@
 package collectd
 
-import "net"
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"time"
 
-type Service struct{}
+	"github.com/influxdb/influxdb/cluster"
+	"github.com/influxdb/influxdb/tsdb"
+	"github.com/kimor79/gollectd"
+)
 
-func NewService(c Config) *Service { return &Service{} }
+// pointsWriter is an internal interface to make testing easier.
+type pointsWriter interface {
+	WritePoints(p *cluster.WritePointsRequest) error
+}
 
-func (s *Service) Open() error    { return nil }
-func (s *Service) Close() error   { return nil }
-func (s *Service) Addr() net.Addr { return nil }
+// Service represents a UDP server which receives metrics in collectd's binary
+// protocol and stores them in InfluxDB.
+type Service struct {
+	Config       *Config
+	PointsWriter pointsWriter
+	Logger       *log.Logger
+
+	wg      sync.WaitGroup
+	err     chan error
+	stop    chan struct{}
+	ln      *net.UDPConn
+	batcher *tsdb.PointBatcher
+	typesdb gollectd.Types
+	addr    net.Addr
+}
+
+// NewService returns a new instance of the collectd service.
+func NewService(c Config) *Service {
+	s := &Service{
+		Config: &c,
+		Logger: log.New(os.Stderr, "[collectd] ", log.LstdFlags),
+		err:    make(chan error),
+	}
+
+	return s
+}
+
+// Open starts the service.
+func (s *Service) Open() error {
+	assert(s.Config.BindAddress != "", "BindAddress is blank")
```

Comment:
It has a default bind address.  If the user enables the collectd service and then specifically overrides the default bind address with blank, seems like that should be a panic.  No?